### PR TITLE
Add support for `Binding<Detent>`

### PR DESCRIPTION
**Pull request by samdogg7**
- [x] Make `BottomSheet` a protocol
- [x] Create `IsPresentedBottomSheet` and `SelectedDetentBottomSheet` that conform to `BottomSheet`
- [x] Add documentation to new changes

### CODE REVIEW

**Comment by samdogg7** [comment]
Ah hold on, my stash removed a lot more than anticipated...

**Comment by samdogg7** [comment]
Working now on re-implementing `isPresented` bindings... Draft for now

**Comment by samdogg7** [comment]
@wojtek717 I managed to support both, in a cleaner manner than I though 😄 Let me know what you think!

File: Sources/BottomSheet/Detent.swift [hunk 1]
```diff
@@ -21,6 +22,8 @@ public enum Detent {
             return 0.50
         case .small:
             return 0.20
+        case .hidden:
+            return 0.0
         case .fraction(let value):
             return value
```
Comment (wkonury) [inline Sources/BottomSheet/Detent.swift:28]: When we are working with Detents, maybe we should check if value is not negative? There is no sense with that value for .fraction

File: Sources/BottomSheet/IsPresentedBottomSheet.swift [hunk 2]
```diff
@@ -0,0 +1,79 @@
+//
+//  IsPresentedBottomSheet.swift
+//  BottomSheet
+//
+//  Created by Sam Doggett on 10/28/24.
+//
+
+import SwiftUI
+
+struct IsPresentedBottomSheet<Content: View, SheetContent: View>: BottomSheet {
```
Comment (wkonury) [inline Sources/BottomSheet/IsPresentedBottomSheet.swift:10]: "IsPresentedBottomSheet" is very odd struct name for me. Why not just BottomSheet and BottomSheetProtocol?

File: Sources/BottomSheet/IsPresentedBottomSheet.swift [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    @State var configuration: BottomSheetViewConfiguration
+
+    @State var selectedDetent: Detent = .small
+
+    @Binding var isPresented: Bool
```
Comment (wkonury) [inline Sources/BottomSheet/IsPresentedBottomSheet.swift:15]:
```suggestion
    @State var configuration: BottomSheetViewConfiguration
    @State var selectedDetent: Detent = .small

    @Binding var isPresented: Bool
```

**Review by wojtek717 — APPROVED** [review]

**Pull request merged**